forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabRepository.java

```diff
@@ -556,10 +556,15 @@ private Optional<CommitComment> findComment(String commitTitle,
         // exists on that commit before we try to process it. If this fails it's
         // most likely due to inconsistent data from GitLab, which should
         // eventually clear up on subsequent tries.
-        Optional<CommitComment> found = candidates.stream()
-                .flatMap(candidate -> commitComments(candidate).stream())
-                .filter(comment -> comment.id().equals(commentId))
-                .findFirst();
+        Optional<CommitComment> found = Optional.empty();
+        for (var candidate : candidates) {
```

Comment:
Are you sure this makes a difference? Streams aren't supposed to be eager, so I was expecting the current implementation to do exactly the same thing.

Reply:
I am not sure what do you mean by eager. I think in the original implementation, it will retrieve all the commit comments before filtering, so with so many candidates, it's inefficient.

Reply:
Ah, I did some tests, you are right, need to think more about it